cloud_run_service_pubsub/main.tf

```diff
@@ -26,16 +26,20 @@ resource "google_service_account" "sa" {
 # [START cloudrun_service_pubsub_run_invoke_permissions]
 resource "google_cloud_run_service_iam_binding" "binding" {
   location = google_cloud_run_service.default.location
-  service = google_cloud_run_service.default.name
-  role = "roles/run.invoker"
-  members = ["serviceAccount:${google_service_account.sa.email}"]
+  service  = google_cloud_run_service.default.name
+  role     = "roles/run.invoker"
+  members  = ["serviceAccount:${google_service_account.sa.email}"]
 }
 # [END cloudrun_service_pubsub_run_invoke_permissions]
 
 # [START cloudrun_service_pubsub_token_permissions]
-resource "google_project_iam_binding" "project" {
+
+data "google_project" "project" {
+}
+
+resource "google_project_iam_binding" "project_token_creator" {
   role    = "roles/iam.serviceAccountTokenCreator"
-  members = ["serviceAccount:${google_service_account.sa.email}"]
+  members = ["serviceAccount:service-${data.google_project.project.number}@gcp-sa-pubsub.iam.gserviceaccount.com"]
 }
```

Comment:
We can use `google_project_service_identity` resource instead to retrieve the pubsub service agent email.